lang/frontend/language/src/main/java/tools/vitruv/neojoin/utils/EMFUtils.java

```diff
@@ -150,4 +158,35 @@ public static String diagnosticSeverityText(Diagnostic diagnostic) {
         }
     }
 
+    /**
+     * Write the given {@link EObject} to a file specified by the given {@link URI}.
+     *
+     * @param uri    URI of the output file
+     * @param object the {@link EObject} to write
+     */
+    public static void save(URI uri, EObject object) throws IOException {
+        save(uri, object, () -> new XMIResourceFactoryImpl());
```

Comment:
The correct resource factory for Ecore meta models is `EcoreResourceFactoryImpl`. I'd suggest omitting the on demand registration and just assert that some factory is registered?